lib/vdc/Grid.cpp

```diff
@@ -139,6 +139,15 @@ void Grid::GetRange(float range[2]) const
     float missingValue = GetMissingValue();
     Grid::ConstIterator itr    = this->cbegin();
     Grid::ConstIterator enditr = this->cend();
+
+	// Edge case: all values are missing values.
+	//
+	range[0] = range[1] = missingValue;
+	while (*itr == missingValue && itr != enditr) {
+		++itr;
+	}
+	if (itr == enditr) return;
+
```

Comment:
Three comments. 

1. This loop can be simplified by making use of an std algorithm: `std::all_of()`. 

2. This fix seems to break the logic of the rest of the function. Namely, if there exists non-missingVal elements, then `itr` will be pointing at an arbitrary location, and you won't find the true min and max. 

3. The code to look for min and max elements could be simplified by making use of another std algorithm: `std::minmax_element()`

Reply:
> Three comments.
> 
> 1. This loop can be simplified by making use of an std algorithm: `std::all_of()`.

The Grid operators unfortunately are not STL C++ operators, so they can't be used with the STL.

> 2. This fix seems to break the logic of the rest of the function. Namely, if there exists non-missingVal elements, then `itr` will be pointing at an arbitrary location, and you won't find the true min and max.

Hmm. Are you sure? I don't see how 'itr' will be pointing at an invalid location. AFter the first check for leading missing values 'itr' is either at the end of the list, or pointing to a value, no?


> 3. The code to look for min and max elements could be simplified by making use of another std algorithm: `std::minmax_element()`

Same comment as above.

Reply:
Well, as long as the grid iterator meets a set of requirements, those iterators can be used in STL algorithms. A compiler can tell you if the iterator you give to it meets those requirements. That being said, it's only a suggestion to use STL algorithms.

Regarding the logic, you're right, it's not going to break anything. Now the PR is ready to merge!